### ISSUE TITLE
Hello, I have a question: why are real_A and real_B spliced ​​together when training unpaired images, but there is no such step when training paired images?

### ISSUE THREAD

**Issue by HNU-CPF** [Yuxinn-J/Scenimefy#10]
Hello, I have a question: why are real_A and real_B spliced ​​together when training unpaired images, but there is no such step when training paired images?
 """
 Run forward pass; called by both functions <optimize_parameters> and <test>."""
        self.real = torch.cat((self.real_A, self.real_B),
                              dim=0) if self.opt.dce_idt and self.opt.isTrain else self.real_A
        if self.opt.flip_equivariance:
            self.flipped_for_equivariance = self.opt.isTrain and (np.random.random() < 0.5)
            if self.flipped_for_equivariance:
                self.real = torch.flip(self.real, [3])

        # forward unpaired data
        self.fake = self.netG(self.real)
        self.fake_B = self.fake[:self.real_A.size(0)]
        if self.opt.dce_idt:
            self.idt_B = self.fake[self.real_A.size(0):]

        # forward paired data
        self.fake_p = self.netG(self.real_A_p)
        self.fake_B_p = self.fake_p[:self.real_A_p.size(0)]

**Comment by Yuxinn-J** [comment]
Hii thanks for reaching out!

The concat step is used to calculate both G(X) and G(Y), and futher computes the hdce identity loss between Y and G(Y), which helps the model capture heterogeneous semantic relations. We omit this loss in the supervised branch to prevent the potiential degradation in performance due to lower quality pseudo data Y_p. If you are interested in the details of the loss applied in the unsupervised branch, please refer to Section 3.3.2 of the main paper.

I hope this clarifies your query!

**Comment by Yuxinn-J** [comment]
Hii since there's no more activity here, I'll close this issue. Feel free to open a new one if you have any questions :)
